--- vendor/src/platform/CMakeLists.txt
+++ vendor/src/platform/CMakeLists.txt
@@ -95,11 +95,6 @@ install(FILES src/util/atomic.h
               src/util/util.h
         DESTINATION include/p8-platform/util)
 
-if(MSVC)
-  # install generated pdb
-  install(FILES $<TARGET_FILE_DIR:p8-platform>/p8-platform.pdb
-          DESTINATION "${CMAKE_INSTALL_LIBDIR}")
-endif(MSVC)
 
 if(NOT WIN32)
   configure_file(p8-platform.pc.in p8-platform.pc @ONLY)
